api/src/main/java/org/apache/iceberg/expressions/ExpressionUtil.java

```diff
@@ -176,6 +223,28 @@ public static String describe(Term term) {
     }
   }
 
+  public static <T> UnboundTerm<T> unbind(BoundTerm<T> term) {
+    if (term instanceof BoundTransform) {
+      BoundTransform<?, T> bound = (BoundTransform<?, T>) term;
+      return Expressions.transform(bound.ref().name(), bound.transform());
+    } else if (term instanceof BoundReference) {
+      return Expressions.ref(((BoundReference<T>) term).name());
+    }
+
+    throw new UnsupportedOperationException("Cannot unbind unsupported term: " + term);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static <T> UnboundTerm<T> unbind(Term term) {
```

Comment:
I think it's useful so I'd probably keep it. Since we are adding the ability to unbind terms, it seems like doing both versions at once is the right thing to do.